### ISSUE TITLE
Map does not show

### ISSUE THREAD

**Issue by DennisKempel** [streetcomplete/StreetComplete#3837]
**How to Reproduce**
Reinstall SC, maybe on a phone that has been reset to factory settings. Experienced it on a Google Pixel 3 and 5.

**Versions affected**
v41.1

**Comment by FloEdelmann** [comment]
Could you please elaborate? Which are the exact steps to reproduce? Did you complete the tutorial? Does it work after restarting the app? Which version of Android? Can you post a screenshot?

**Comment by DennisKempel** [comment]
![Screenshot_20220304-092444](https://user-images.githubusercontent.com/65849470/156726865-42d0f506-8059-48ce-8a81-d5993bdd8910.png)

Reproduce: After installing the app on a blank Android V12 system, latest patch level.

Reinstallation/restarting does not help.

**Comment by matkoniecz** [comment]
Has it worked with previous SC version?

Is uninstalling and installing one of older versions from https://github.com/streetcomplete/StreetComplete/releases works?

Is reinstalling SC several hours later solving issue? Maybe it is just hiccup of tile servers.

**Comment by DennisKempel** [comment]
yes, it worked on previous versions. I was also able to solve it on my other mobile by closing the app and zooming in and out and at some point it worked again. 

I am now back on my old phone (Pixel 3), but ne of the methods work yet to bring the map back. I installed the system yesterday afternoon, so the timed cache does not seem to be the problem, too (?)

**Comment by matkoniecz** [comment]
If it used to work: is uninstalling and installing one of older versions from https://github.com/streetcomplete/StreetComplete/releases works?

That would allow to check is it something that become changed in SC caused problems (unlikely, but worth checking), or has situation changed elsewhere.

**Comment by mnalis** [comment]
Also, trying to manually open map at https://maps.jawg.io/ in your _web browser on the affected phones_ might show if the problem is specific to StreetComplete, or if it is networking / firewall related.

**Comment by DennisKempel** [comment]
trying to open maps.jawg.io gives

ERR_SSL_PROTOCOL_ERROR

if I turn off WiFi it works. I am not aware there is any special firewall limitation in my router besides usual security restrictions. Any idea what could cause this?

**Comment by westnordost** [comment]
Usually your browser should give you an option to view what is exactly the problem with the secured connection. Revoked certificate? Not trusted? Old?

In any case, suffice to say this happened before for someone else, though it was unclear where the issue was.

![image](https://user-images.githubusercontent.com/4661658/156786693-00c2ba33-a986-475d-978f-ba58e97d4d3b.png)

**Comment by DennisKempel** [comment]
Back on Wifi I can still see the map. However, if I delete the chrome mobile cache I get the error again. 

Chrome tells me "site is insecure" and "page identity cannot be verified". Further details as given by chrome: https://support.google.com/chrome/answer/95617?hl=en

As far as I know this is rather a misconfiguration error on server side, in this case jawg.io. Could it be their certificate does not cover their subdomain maps.jawg.io?

**Comment by mnalis** [comment]
@DennisKempel I don't think this is related to error `@westnordost` posted (which seems to be timeout - usually due to network interactivity problems at ISP or its upstream, or DNS problems, which happens all the time but it usually quickly corrected)

> trying to open maps.jawg.io gives
> ERR_SSL_PROTOCOL_ERROR

That error  indicates that the version of `https://` that web site thinks your browser supports (which need not be the same as what your browser _actually supports_, see below) is not supported by web site. This should never happen unless either the website is very old and unmaintained (which it is not), or your browser/app is very old and unmaintained (which it is not), or some party in the middle is intercepting your connections and it is very old and unmaintained (likely).

<details>

When I lookup from here, maps.jawg.io supports modern `https` version `TLS1.3` and `TLS1.2` only, and it does not supported older `TLS1.1`, `TLS1.0`, `SSL3`, `SSL2`: https://www.ssllabs.com/ssltest/analyze.html?d=maps.jawg.io

So, if you browser for example only supported `TLS1.0` and `TLS1.1` it wouldn't be able to connect and would fail with error you mention above.

However, as you say you use Android 12 (hopefully updated), it should support modern `TLS1.3` / `TLS1.2` ciphers, so I'd be quite surprised if your browser was too old to support modern TLS protocols.

</details>

> if I turn off WiFi it works. 

ah, this is **_very_** indicative. 

You can test what protocols your browser support by going to https://www.ssllabs.com/ssltest/viewMyClient.html
See what is says when you are connected via WiFi, **and** what when you're connected via mobile data.

I'd suspect it might differ depending on network connection you choose. 

<details>

It indicates [MiTM](https://en.wikipedia.org/wiki/Man-in-the-middle_attack). This is when **_something_** between your browser and destination web server intercepts/hijacks the `https` connection; usually in order to sniff what you're doing. While it generally can be some malware on your phone or software installed by company on employee phone to monitor what employee does to prevent data leaks, it is also unfortunately sometimes marketed as "child / content protection" feature, which in order to filter out pages containing "inappropriate" content needs to hijack encrypted `https` connection (otherwise,it wouldn't know what content the user is attempting to view!)

</details> 

>I am not aware there is any special firewall limitation in my router besides usual security restrictions. Any idea what could cause this?

I'd guess either your WiFi router itself (most likely), or ISP behind it, is doing such "child protection" MiTM connection hijacking; and the software doing that is old enough not to support modern TLS protocols, so it fails to connect at all. Workaround would be looking in your router settings and **disabling such "inappropriate content protection"** to see if it fixes it. If that fixes it, you can either live with it disabled or try upgrading your router software or asking for newer router; and if disabling such features on your wifi router does not help, contact the ISP and ask them about such features.

<details>

Alternative (much less likely) explanations why it works on some network connection and not the other might be:
- someone is trying to hack DNS of your ISP or of  jawg.io to maliciously intercept that traffic (unlikely, as not really useful target, and others would probably have noticed by now)
- malware on your phone which only happens on one connection (quite unlikely, as it happens on multiple phones, and malware usually works just fine on any network connection)
- if company phones, "guardian" software installed by company
- `maps.jawg.io` name resolves to different Jawg server in some cases, and only some of those servers are unmaintained (unlikely).

</details>


**TL;DR:** disable child / content protection "security" feature on your router

**Comment by westnordost** [comment]
That details page does not explain what the error is. You have to click on "Advanced" ("Erweitert") on the error message in Chrome. Then, you'll see the error message.

![image](https://user-images.githubusercontent.com/4661658/156819617-e320c43e-b646-4736-84e5-19265adb124a.png)

**Comment by DennisKempel** [comment]
Protocol Support
Your user agent has good protocol support.
Your user agent supports TLS 1.2 and TLS 1.3, which are recommended protocol version at the moment.

The router is a Fritz box 7490 with all current updates that I maintain myself. I'm using German Telekom, so I doubt there is any "old" software.

My browser gives no additional information. I do not have the "Erweitert" button. If I click the "i" on the URL box I see what is on the screenshot.
![Screenshot_20220304-193949](https://user-images.githubusercontent.com/65849470/156823190-d92830ec-34f6-4a48-84f5-4c63c2e2237d.png)

**Comment by DennisKempel** [comment]
I think this indicates a misconfiguration on server side...

**Comment by westnordost** [comment]
So you don't have the same issue on your desktop PC with maps.jawg.io?

**Comment by DennisKempel** [comment]
...looks I was wrong. When I browse maps.jawg.io in incognito mode I see a note from my Fritz box router saying that some "Internet filter" prevents maps.jawg.io from being shown.

Since I have not prepared my own list this may mean that maps.jawg.io is on some public listing. Let me try to look it up...

**Comment by DennisKempel** [comment]
...however this got in my router it is obviously the reason for my issue:

![Screenshot_20220304-201147](https://user-images.githubusercontent.com/65849470/156826844-e63f4a6e-9766-4a5c-8bd7-dad8123b567b.png)

Having removed the entry the site works. Sorry for the troubles.

**Comment by westnordost** [comment]
Did you put that there? Why?

**Comment by DennisKempel** [comment]
not sure, I don't remember it.

It may be worth it if maps.jawg.io cannot be reached by SC to display some sort of (error) message, anyway this seems to having been my own mistake.
